src/ppx_deriving_yojson.ml

```diff
@@ -609,9 +609,18 @@ let desu_str_of_type ~options ~path ({ ptype_loc = loc } as type_decl) =
             Exp.case
               [%pat? `List ((`String [%p pstr name]) ::
                                      [%p plist (List.mapi (fun i _ -> pvar (argn i)) args)])]
-              (desu_fold ~quoter ~loc ~path (fun x -> constr name' x) args)
+              (desu_fold ~quoter ~loc ~path
+                 (fun x ->
+                    (* https://github.com/ocaml-ppx/ppx_deriving_yojson/issues/163 *)
+                    let typ = Ppx_deriving.core_type_of_type_decl type_decl in
+                    Exp.constraint_ (constr name' x) typ)
```

Comment:
Same here, we're going to deprecate `Ast_helper` in favor of `Ast_builder` in ppxlib so I'd suggest instead going for:
```suggestion
                    Ast_builder.Default.pexp_constraint ~loc (constr name' x) typ)
```